IOS/SwiftSeedProject/SwiftSeedProject/Application/ViewModels/LoginViewModel.swift

```diff
@@ -0,0 +1,79 @@
+//
+//  ViewModel.swift
+//  SwiftSeedProject
+//
+//  Created by Brian Sztamfater on 27/3/17.
+//  Copyright © 2017 Making Sense. All rights reserved.
+//
+
+import Bond
+
+class LoginViewModel {
+    
+    // TODO: Create a service for authentication and remove references to Api
+    private let api: Api
+    private var user: User?
+    
+    var name = Observable<String?>("")
+    var email = Observable<String?>("")
+    
+    init() {
+        // TODO: We need to implement dependency injection
+        self.api = Api.init()
+    }
+    
+    func getRandomUser() {
+        let randomId:UInt32 = arc4random_uniform(10) + 1; // Fake API has users with id from 1 to 10
+        api.getUser(userId: String(randomId)) { [weak self] result in
+            do {
+                guard let strongSelf = self else {
+                    return
+                }
+                strongSelf.user = try result.unwrap()
+                strongSelf.name.value = strongSelf.user!.name
+                strongSelf.email.value = strongSelf.user!.email
+                debugPrint("Got user with id: " + strongSelf.user!.id!)
+            } catch let error as NSError {
+                debugPrint("Get user error: \(error.localizedDescription)")
+            }
+        }
+    }
+    
+    func updateUser() {
+        if self.user != nil {
+            self.user!.name = self.name.value!; // Should we bind the model with the viewModel?
+            self.user!.email = self.email.value!;
+            api.createUser(user: self.user!) { [weak self] result in
+                do {
+                    guard let strongSelf = self else {
+                        return
+                    }
+                    strongSelf.user = try result.unwrap()
```

Comment:
WET code..

(Violations of DRY are typically referred to as WET solutions, which is commonly taken to stand for either "write everything twice", "we enjoy typing" or "waste everyone's time" - [source](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself))

You asked for a thorough review 😄 

This block is almost identical to the one above.